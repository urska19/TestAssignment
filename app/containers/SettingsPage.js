import React, {Component} from 'react';
import {Text, View, TouchableOpacity, ListView, ScrollView} from 'react-native';
import ContainerRowView from '../components/ContainerRowView';
import AppSettings from '../config/AppSettings';
import {observable} from 'mobx';
import { observer } from 'mobx-react/native';
import styles from './styles/SettingsPageStyle';
import GlobalStore from '../store/globalStore';

@observer
export default class SettingsPage extends React.Component {
  
  constructor (props) {
    super(props);
  }

  currencyList() {
    var list = [];
    for(var item in AppSettings.availableCurrencies) {
      list.push(this.currencyListElement(item, AppSettings.availableCurrencies[item]))
    }
    return (
      list
    );
  }

  currencyListElement(key, item) {
    return (
      <ContainerRowView
        key={key}
        listRowContainerStyle={[styles.listRowContainerStyle, {backgroundColor: key == GlobalStore.currentlySelectedCurrency ? 'yellow' : 'white'}]}
        onPress={(text) => {GlobalStore.currentlySelectedCurrency = key}}
        insideElement={this.renderTextComponent(key, item)}
      />
    )
  }

  renderTextComponent(key, item) {
    return (
      <Text style={styles.textStyle}>{item.name}</Text>
    )
  }

  render () {
    this.currencyList()
    return (
      <View style={styles.container}> 
        <View style={styles.innerContainer}> 
          {this.currencyList()}
        </View>
      </View>
    );
  }

}
