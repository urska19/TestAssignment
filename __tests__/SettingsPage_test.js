import 'react-native';
import React from 'react';
import SettingsPage from '../app/containers/SettingsPage';
import renderer from 'react-test-renderer';
jest.mock('mobx-react/native', () => require('mobx-react/custom'));

it('renders correctly', () => {
  const tree = renderer.create(
    <SettingsPage />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});