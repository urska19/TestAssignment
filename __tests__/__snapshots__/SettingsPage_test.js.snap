// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "gray",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "marginHorizontal": 10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 46,
          "justifyContent": "space-between",
          "marginVertical": 2,
          "opacity": 1,
          "paddingHorizontal": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        US Dollar
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "yellow",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 46,
          "justifyContent": "space-between",
          "marginVertical": 2,
          "opacity": 1,
          "paddingHorizontal": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Euro
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 46,
          "justifyContent": "space-between",
          "marginVertical": 2,
          "opacity": 1,
          "paddingHorizontal": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Chinese Yuan
      </Text>
    </View>
  </View>
</View>
`;
